sympy/core/trace.py

```diff
@@ -1,4 +1,10 @@
-from sympy import Expr, Add, Mul, Pow, sympify, Matrix, Tuple
+from .add import Add
+from .containers import Tuple
+from .expr import Expr
```

Comment:
This deprecation decorator wasn't actually firing. The problem is that the @deprecated decorator was designed to add an `__init__` when added to a class. But because this class never returns itself, it never calls `__init__`. 

I'll look at making this work better, but a much simpler way to do this would be to just deprecate the entire module at import, and just add `from sympy.physics.quantum.trace import Tr`. The only issue with that is that module imports can only be run once, but that isn't a huge problem. 

I'm fixing this in https://github.com/sympy/sympy/pull/22900